src/lib/formatting/redact.mjs

```diff
@@ -0,0 +1,50 @@
+/**
+ * Redacts a string by replacing everything except the last four characters with asterisks.
+ * If the string is less than 12 characters long, it is completely replaced with asterisks.
+ *
+ * @param {string} text - The string to redact.
+ * @returns {string} The redacted string.
+ */
+export function redact(text) {
+  if (!text) return text;
+
+  if (text.length < 12) {
+    return "*".repeat(text.length);
+  }
+
+  const lastFour = text.slice(-4);
+  const redactedLength = text.length - 4;
+  return "*".repeat(redactedLength) + lastFour;
```

Comment:
might be good to put some of the prefix in too as they can hint to us what type of key it is.

See: https://github.com/fauna/backup-restore-frontdoor/blob/main/src/auth/constants.ts

For some examples

Reply:
Honestly, i'd rather just do the prefix and redact all the rest. that way none of the random data is displayed. its just the envelope flag.

Reply:
That won't help a user figure out what secret was used if it's not something they expect (such as an expected profile or env passing secret). How about for strings larger than 16, we keep the first and last 4 and redact the middle?